File: app/src/main/java/org/baiyu/fuckkiller/SettingsActivity.java
package org.baiyu.fuckkiller;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.text.InputType;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.EditTextPreference;
import androidx.preference.PreferenceFragmentCompat;

import de.robv.android.xposed.XposedBridge;

public class SettingsActivity extends AppCompatActivity {
    /** @noinspection deprecation*/
    @SuppressLint("WorldReadableFiles")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_layout);
        try {
            getSharedPreferences(BuildConfig.APPLICATION_ID + "_preferences", MODE_WORLD_READABLE);
        } catch (Exception e) {
            XposedBridge.log(e);
        }
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settings_container, new MySettingsFragment())
                .commit();
    }

    private static class MySettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(@Nullable Bundle savedInstanceState, @Nullable String rootKey) {
            setPreferencesFromResource(R.xml.preferences, rootKey);

            EditTextPreference MAX_PHANTOM_PROCESSES = findPreference("MAX_PHANTOM_PROCESSES");
            EditTextPreference MAX_CACHED_PROCESSES = findPreference("MAX_CACHED_PROCESSES");

            for (EditTextPreference p : new EditTextPreference[]{MAX_CACHED_PROCESSES, MAX_PHANTOM_PROCESSES}) {
                if (p != null) {
                    p.setSummaryProvider(EditTextPreference.SimpleSummaryProvider.getInstance());
                    p.setOnBindEditTextListener(editText -> editText.setInputType(InputType.TYPE_CLASS_NUMBER));
                }
            }
        }
    }
}

File: app/src/main/kotlin/org/lyaaz/fuckkiller/SettingsActivity.kt
package org.lyaaz.fuckkiller

import android.annotation.SuppressLint
import android.os.Bundle
import android.text.InputType
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.EditTextPreference
import androidx.preference.PreferenceFragmentCompat

class SettingsActivity : AppCompatActivity() {
    /**
     * @noinspection deprecation
     */
    @SuppressLint("WorldReadableFiles")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_layout)
        try {
            @Suppress("DEPRECATION")
            getSharedPreferences(BuildConfig.APPLICATION_ID + "_preferences", MODE_WORLD_READABLE)
        } catch (e: Exception) {
            getSharedPreferences(BuildConfig.APPLICATION_ID + "_preferences", MODE_PRIVATE)
        }
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.settings_container, MySettingsFragment())
            .commit()
    }

    private class MySettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.preferences, rootKey)

            val maxPhantomProcesses = findPreference<EditTextPreference>("MAX_PHANTOM_PROCESSES")
            val maxCachedProcesses = findPreference<EditTextPreference>("MAX_CACHED_PROCESSES")

            listOf(maxCachedProcesses, maxPhantomProcesses).forEach {
                it?.apply {
                    summaryProvider = EditTextPreference.SimpleSummaryProvider.getInstance()
                    setOnBindEditTextListener { editText: EditText ->
                        editText.inputType = InputType.TYPE_CLASS_NUMBER
                    }
                }
            }
        }
    }
}
